primitives/teerex/src/lib.rs

```diff
@@ -52,3 +56,26 @@ pub struct Request {
 	pub shard: ShardIdentifier,
 	pub cyphertext: Vec<u8>,
 }
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, Copy, Default, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Header {
+	/// The parent hash.
+	pub parent_hash: H256,
+
+	/// The block number.
+	pub block_number: u64,
+
+	/// The Shard id.
+	pub shard_id: ShardIdentifier,
+
+	/// The payload hash.
+	pub block_data_hash: H256,
+}
```

Comment:
Is this a pattern that we have for other data structures as well, i.e. duplicating code of the worker?

Reply:
Yes, we currently are duplicating a lot. Before bickering about this extra duplication, we should define where we would want a shared primitives crate: #50